frontend/src/utils/api.js

```diff
@@ -141,10 +142,13 @@ async function deleteAndRefreshResource(id) {
 }
 
 async function filterAndRefreshResource(keyword, address, tag, radius) {
+  store.dispatch(updateSearchParams({ keyword, address, tag }));
   const results = await getSearchResults(keyword, address, tag, radius);
   store.dispatch(updateResources(results.resources));
-  if (results.center) {
+  if (results.center && results.center[0]) {
     store.dispatch(updateMapCenter(results.center));
+  } else {
+    store.dispatch(updateMapCenter(null));
```

Comment:
Nice subtle catch here—I see how it's much better now to store `null` here for the value rather than an array with `[null, null]`